### ISSUE TITLE
AddOptionalParameterToInvocationCodeRefactoringProvider index out of range

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7045]
Put the caret on a local variable and the info bar showed an error.

VS Mac: 8.0.0.1344
Git revision: b7efc0cb39fe50ffde40506da2df3d62f11a8528
Build date: 2019-02-03 16:55:27+00
Build branch: master
Xamarin extensions: 956d4c6701a2c8224a2efe99de3cb91176423155

```
ERROR [2019-02-04 20:28:23Z]: Roslyn error: Extension_Exception AddOptionalParameterToInvocationCodeRefactoringProvider : Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.Collections.Generic.List`1[T].RemoveAt (System.Int32 index) [0x00071] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:930 
  at RefactoringEssentials.CSharp.CodeRefactorings.AddOptionalParameterToInvocationCodeRefactoringProvider.ComputeRefactoringsAsync (Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContext context) [0x0039b] in /Users/vsts/agent/2.146.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Custom/AddOptionalParameterToInvocationCodeRefactoringProvider.cs:84 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan state, Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider provider, Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, System.Threading.CancellationToken cancellationToken) [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2019-02-04 20:28:30Z]: Roslyn reported an exception to the user
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.Collections.Generic.List`1[T].RemoveAt (System.Int32 index) [0x00071] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:930 
  at RefactoringEssentials.CSharp.CodeRefactorings.AddOptionalParameterToInvocationCodeRefactoringProvider.ComputeRefactoringsAsync (Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContext context) [0x0039b] in /Users/vsts/agent/2.146.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Custom/AddOptionalParameterToInvocationCodeRefactoringProvider.cs:84 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan state, Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider provider, Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, System.Threading.CancellationToken cancellationToken) [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2019-02-04 20:28:37Z]: Roslyn reported an exception to the user
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.Collections.Generic.List`1[T].RemoveAt (System.Int32 index) [0x00071] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:930 
  at RefactoringEssentials.CSharp.CodeRefactorings.AddOptionalParameterToInvocationCodeRefactoringProvider.ComputeRefactoringsAsync (Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContext context) [0x0039b] in /Users/vsts/agent/2.146.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Custom/AddOptionalParameterToInvocationCodeRefactoringProvider.cs:84 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync (Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan state, Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider provider, Microsoft.CodeAnalysis.Extensions.IExtensionManager extensionManager, System.Threading.CancellationToken cancellationToken) [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

> VS bug [#785478](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/785478)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-8] Comment by **xamarinc**



Fixed in version 8.0.0.1394 (master)



Pull Request #7070 merged by: Marius Ungureanu  
 Author: mono  
 Commit: 07910f4115ac401ff3aecfb44fc46e27c985c488 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-10] Comment by **xamarinc**



Fixed in version 8.0.0.2001 (release-8.0)



Author: Marius Ungureanu  
 Commit: 892364d64238a07f4da72b2b411b46f1d65c0e2b (mono/RefactoringEssentials)  
 Included in Commit: 9d6d103016ef146fd88637492789a4df947959dc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-12] Comment by **xamarinc**



Fixed in version 8.0.0.2176 (release-8.0)



Pull Request #7070 merged by: Marius Ungureanu  
 Author: mono  
 Commit: 07910f4115ac401ff3aecfb44fc46e27c985c488 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7070 merged by: Marius Ungureanu  
 Author: mono  
 Commit: 07910f4115ac401ff3aecfb44fc46e27c985c488 (xamarin/monodevelop)
